Clarify undefined-behavior entry for concurrent use of teams (#426)
openshmem-org/specification: content/backmatter.tex

\end{itemize}
\tabularnewline
\hline
Multithreaded use of a team in concurrent team-based collectives &
Team-based collective operations are not thread-safe on the same \VAR{team}
object.

> Concurrent use of a team from multiple threads may result in undefined

Reviewer: ```suggestion
Concurrent collective operations on the same team from multiple threads may result in undefined
```
